### ISSUE TITLE
Exploit: You can look at NPCs while blind

### ISSUE THREAD

**Issue by NetSysFire** [CleverRaven/Cataclysm-DDA#55803]
### Describe the bug

See title.

### Steps to reproduce

1. Put on a blindfold or be blind in another way.
2. Size up stats (only intelligence might make sense to be "visible" to you while blind) or look at the equipment of the NPC etc

### Expected behavior

You get a message that you can not see.

There should be a check in the code for blindness or high eye encumbrance (maybe, might be out of scope here) and alternative messages, e.g "You can't see what <NPC> is wearing!"

There should perhaps also be a check when you ask the NPC to teach you things. Teaching you social skills may work while blind, but this does not apply to combat or shooting or anything else, really.

Adding a check for blindness should be easy enough for new contributors. I considered adding Low Priority to this issue because this is not gamebreaking and more of an edge case.

### Versions and configuration

- experimental

**Comment by wuteek** [comment]
Hello I'm interested in working on this! Are you able to assign it to me?

**Comment by Night-Pryanik** [comment]
There's no need in assignments. Just make your changes and submit a PR.

**Comment by Night-Pryanik** [comment]
How are you able to talk to the NPC while being blind? The sight range is 0, so there's no option to talk to anyone because you can't see anything.

**Comment by LiramuIorin** [comment]
Just a heads up that I'm working on this and already fixed this exploit, at least for spawned npcs. I ran out of time today, but I'll be submitting a pull request tomorrow at the latest.

**Comment by NetSysFire** [comment]
Did you see the already existing mentioned PR?

**Comment by LiramuIorin** [comment]
Nope. My bad, carry on.

**Comment by NetSysFire** [comment]
No its totally fine if you work on this, too, I was more concerned about that wasting your time. The linked PR is by @Night-Pryanik and has not seen much recent activity, so perhaps there is some help needed and you could cooperate.

**Comment by Night-Pryanik** [comment]
@LiramuIorin please continue your work and share it with us. Maybe your solution is better than mine.

**Comment by LiramuIorin** [comment]
Alright, I made the pull request. Honestly, as this is my first contribution, I expect I made some mistakes somewhere. And my fix was very basic, simply adding is_blind checks in one file and attached text.

**Comment by JazzGlobal** [comment]
Can this issue be closed? Looks like it has been resolved.

**Comment by Night-Pryanik** [comment]
I don't think so. There are other issues still unresolved, for example NPCs teaching blind player.
